### ISSUE TITLE
yodaConditions rule breaks with keypaths

### ISSUE THREAD

**Issue by kevinrenskers** [nicklockwood/SwiftFormat#397]
Please see this diff:

```
-      return try CampaignUser.query(on: request).filter(\.userID == user.requireID()).filter(\.campaignID == campaign.requireID()).first().flatMap(to: Response.self) { campaignUser in
+      return try CampaignUser.query(on: request).filter(\user.requireID() == .userID).filter(\campaign.requireID() == .campaignID).first().flatMap(to: Response.self) { campaignUser in
```

**Comment by nicklockwood** [comment]
@kevinrenskers thanks for reporting it, I'll get that fixed asap.

**Comment by nicklockwood** [comment]
FYI this is now fixed on the [develop](https://github.com/nicklockwood/SwiftFormat/tree/develop) branch.

**Comment by nicklockwood** [comment]
@kevinrenskers fixed in 0.40.4.
